src/Workspaces/Core/Portable/Rename/Renamer.IRenameAction.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    public static partial class Renamer
+    {
+        internal interface IRenameAction
```

Comment:
I'm a bit confused on what the goal is here -- I get that the type wasn't publicly derivable, so the 'abstract' was a bit strange from that perspective, but it's otherwise harmless I'd say. So we're now making it concrete, but we're adding a layer of indirection to _keep_ the OOP behavior internally.

This feels like the cure is worse than the disease to me, unless I'm missing something...?

Reply:
If nothing else, I feel like this needs a good comment on this interface if we do go ahead with this, because I could easily imagine somebody coming by and refactoring this to the current form.

Reply:
This seemed like the best way to have a sealed public class and just keep whatever implementation details we want internal.  Almost like what we do with external access layer. There's some nuance with the implementation being in code style layer as well. 

More than happy to add a comment on the interface to clarify. The current abstract class seems misleading but it could be alright I guess? I think either current form or proposed in the PR are our two options. I still think this PR has a better public facing representation at the cost of a layer of indirection internally.

Reply:
note: i'm fine with the new structure.  though i'm not sure it really seems like it was necessary if the only goal was to expose a new GetDescription method on the API.

Reply:
It just changed from abstract. It was already exposed